### ISSUE TITLE
Invalid codepoints don't raise RangeError on `#chr`

### ISSUE THREAD

**Issue by yous** [jruby/jruby#1921]
I'm using JRuby-1.7.13.

http://en.wikipedia.org/wiki/UTF-8#Invalid_byte_sequences

> Not all sequences of bytes are valid UTF-8. A UTF-8 decoder should be prepared for:
> - A 4-byte sequence (starting with 0xF4) that decodes to a value greater than U+10FFFF

In Ruby 1.9.3,

``` irb
>> 0x110000.chr(Encoding::UTF_8)
RangeError: invalid codepoint 0x110000 in UTF-8
    from (irb):3:in `chr'
    from (irb):3
    from /Users/yous/.rvm/rubies/ruby-1.9.3-p547/bin/irb:12:in `<main>'
```

In JRuby,

``` irb
>> 0x110000.chr(Encoding::UTF_8)
=> "\xF4\x90\x80\x80"
```

http://en.wikipedia.org/wiki/UTF-8#Invalid_code_points

> According to the UTF-8 definition (RFC 3629) the high and low surrogate halves used by UTF-16 (U+D800 through U+DFFF) are not legal Unicode values, and their UTF-8 encoding should be treated as an invalid byte sequence.

In Ruby 1.9.3,

``` irb
>> 0xD800.chr(Encoding::UTF_8)
RangeError: invalid codepoint 0xD800 in UTF-8
    from (irb):1:in `chr'
    from (irb):1
    from /Users/yous/.rvm/rubies/ruby-1.9.3-p547/bin/irb:12:in `<main>'
>> 0xDFFF.chr(Encoding::UTF_8)
RangeError: invalid codepoint 0xDFFF in UTF-8
    from (irb):2:in `chr'
    from (irb):2
    from /Users/yous/.rvm/rubies/ruby-1.9.3-p547/bin/irb:12:in `<main>'
```

In JRuby,

``` irb
>> 0xD800.chr(Encoding::UTF_8)
=> "\xED\xA0\x80"
>> 0xDFFF.chr(Encoding::UTF_8)
=> "\xED\xBF\xBF"
```

**Comment by yous** [comment]
Still reproducable with JRuby-1.7.19 and JRuby-9.0.0.0.pre1.

**Comment by k77ch7** [comment]
@yous this works in 9.1.9.0 and 9.1.10.0-SNAPSHOT (2.3.3) 2017-05-20 7d0ecb9.

**Comment by yous** [comment]
I can confirm that this was fixed in JRuby-9.0.0.0.pre2, maybe by https://github.com/jruby/jruby/commit/f90bcc95884a10259b2625ccfbe91a6ae59e9442. Thank you, @k77ch7.
